### ISSUE TITLE
Bad error message when using the pattern option without specifying a module via the --module option

### ISSUE THREAD

**Issue by MirceaS** [kframework/kore#2254]
The current error message reads
```
The main module, '', was not found. Check the --module flag.
```
What it should read is
```
You must use --module with --pattern.
```

**Comment by ttuegel** [comment]
The problem is, `--module` is optional in the sense that it is required only if `--pattern` is given. We should change the parser so that `--module` and `--pattern` must appear together if they appear at all.

**Comment by ttuegel** [comment]
I started to write how to do this, but it would be much nicer if we just used `optparse-applicative` the way it was intended.

Let's add a new executable called `kore-parse` with two sub-commands, `kore-parse definition` and `kore-parse pattern`. The second command will have the non-optional arguments. There is a [`command`](https://hackage.haskell.org/package/optparse-applicative-0.16.0.0/docs/Options-Applicative.html#v:command) parser for just this case. There is a more detailed example in the [tutorial](https://github.com/pcapriotti/optparse-applicative#commands).
